addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/generators/HeadersGeneratorUnitTest.java

```diff
@@ -391,9 +404,47 @@ void shouldGenerateAllSupportedHeaders() {
                 headers,
                 contains(
                         header("accept", "text/plain"),
-                        header("content-type", "application/json"),
                         header(headerName, headerValue),
-                        header("cookie", cookieName + "=" + cookieValue)));
+                        header("cookie", cookieName + "=" + cookieValue),
+                        header("content-type", "application/json")));
+    }
+
+    @Test
+    void shouldGenerateExpecteddHeaders() {
+        // Given
+        Operation operation = mock(Operation.class);
+
+        ApiResponse response = mockResponseWithMediaTypes("text/plain");
+        mockOperationWithResponses(operation, response);
+
+        RequestBody request = mockRequestWithMediaTypes("multipart/form-data");
+        mockOperationWithRequest(operation, request);
+
+        String headerName = "Content-Type";
+        String headerValue = "multipart/form-data";
+        Parameter headerParameter = headerParam(headerName);
+        given(dataGenerator.generate(headerName, headerParameter)).willReturn(headerValue);
+
+        List<Parameter> parameters = asList(headerParameter);
+        mockOperationWithParameters(operation, parameters);
+
+        OperationModel operationModel = mock(OperationModel.class);
+        given(operationModel.getOperation()).willReturn(operation);
+        // When
+        List<HttpHeaderField> headers =
+                headersGenerator.generate(operationModel, MULTIPART_POST_BODY);
+        // Then
+        assertThat(
+                headers,
+                contains(
+                        header("accept", "text/plain"),
+                        header(
+                                headerName,
+                                headerValue), // This one gets clobbered when the HttpMessage is
```

Comment:
Add the comment before the line, though having the deny list would avoid the need for clarifications.